lib/datadog/core/telemetry/v1/app_started.rb

```diff
@@ -22,6 +22,27 @@ def initialize(additional_payload: nil, configuration: nil, dependencies: nil, i
             @dependencies = dependencies
             @integrations = integrations
           end
+
+          def to_h
+            {}.tap do |app_started_hash|
+              app_started_hash[:additional_payload] = map_hash(@additional_payload) if @additional_payload
+              app_started_hash[:configuration] = map_hash(@configuration) if @configuration
+              app_started_hash[:dependencies] = map_array(@dependencies) if @dependencies
+              app_started_hash[:integrations] = map_array(@integrations) if @integrations
+            end
+          end
```

Comment:
Suggestion: according to #initialize, the fields stored by this class are arrays (or nil).

Consider storing them always as arrays, even if empty. I think this would simplify dealing with them overall, for instance I think the `#to_h` would be simplified as well.

Also Ruby tip: the `Kernel#Array()` function (yes, it's a function named `#Array`, it's not the class `Array`, can be used to very easily transform inputs, even when nil, into an Array:

```ruby
a = Array([1, 2])
# => [1, 2]
> b = Array(nil)
# => []
```